src/codegeneration/LanguageModeTransformerTrait.js

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strong';
+
+import {FUNCTION_BODY} from '../syntax/trees/ParseTreeType.js';
+import {
+  SLOPPY_MODE,
+  STRICT_MODE,
+  STRONG_MODE,
+} from '../syntax/Parser.js';
+
+/**
+ * This function is used as trait to generate a class that that keeps track of
+ * the language mode.
+ *
+ * Usage:
+ *
+ *  class MyTransformer extends
+ *      LanguageModeTransformerTrait(ParseTreeTransformer) {
+ *    ...
+ *  }
+ *
+ * @param {Function} superClass A class that extends ParseTreeTransformer.
+ * @return {Function}
+ */
+export function LanguageModeTransformerTrait(superClass) {
+  return class extends superClass {
+    constructor(...args) {
+      super(...args);
+      this.languageMode_ = SLOPPY_MODE;
+    }
+
+    updateLanguageMode_(list) {
```

Comment:
Consider `statements`; it isn't completely accurate, but `list` is even less so.